core/src/main/java/org/apache/calcite/adapter/enumerable/EnumerableMergeJoin.java

```diff
@@ -131,8 +160,35 @@ public static boolean isMergeJoinSupported(JoinRelType joinType) {
           required, ImmutableList.of(
           required.replace(leftCollation),
           required.replace(rightCollation)));
+    } else if (isSubset(reqKeys, rightKeys)) {
+      // if sort keys are subset of right join keys, we can extend collations to make sure all join
+      // keys are sorted.
+      collation = extendCollation(collation, rightKeys);
+      RelCollation rightCollation = RelCollations.shift(collation,
+          -left.getRowType().getFieldCount());
+      Mappings.TargetMapping invMapping = mapping.inverse();
+      RelCollation leftCollation = RexUtil.apply(invMapping, rightCollation);
+      return Pair.of(
+          required, ImmutableList.of(
+              required.replace(leftCollation),
+              required.replace(rightCollation)));
+    } else if (isPrefixOrderingNotRequired(rightKeys, reqKeys)
+        && allElementsGe(reqKeys, getLeft().getRowType().getFieldCount())) {
```

Comment:
can we use `reqKeys.stream().allMatch()`?

Reply:
Done! Thanks!